runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -26,245 +26,313 @@
 		overlayManager.remove(playerIndicatorsTileOverlay);
 		overlayManager.remove(playerIndicatorsMinimapOverlay);
-		overlayManager.remove(friendMinimapOverlay);
 	}
-
+	
+	private ArrayList<String> callers =	new ArrayList<>();
+	private List<String> pileList;
+	
 	@Subscribe
-	public void onInteractChanged(InteractChanged event)
+	public void onConfigChanged(ConfigChanged e)
 	{
-		Actor actor = event.getActor();
-		if (actor != null
-				&& actor.getName() != null
-				&& isHighlighted(actor))
+		if (config.callers() != null && !config.callers().trim().equals(""))
 		{
-			highlightedPlayers.put(actor.getName().toLowerCase(), actor.getInteracting());
+			getCallerList();
 		}
 	}
-
+	
 	@Subscribe
-	public void onConfigChanged(ConfigChanged event)
+	public void onGameTick(GameTick gameTick)
 	{
-		if (event.getGroup().equals("playerindicators") && event.getKey().equals("highlightedNames"))
+		if (config.highlightPile() && callers != null)
 		{
-			updateHighlightList();
+			for (Player p : client.getPlayers())
+			{
+				for (String name:callers)
+				{
+					Actor pile = null;
+					String finalName = name.toLowerCase().replace("_", " ");
+					if (p.getName().toLowerCase().replace("_", " ").equals(finalName))
+					{
+						pile = p.getInteracting();
+						if (pile != null)
+						{
+							pileList.set(callers.indexOf(name), pile.getName());
+							//pileList.add(pile.getName());
+						}
+						else
+						{
+							pileList.set(callers.indexOf(name), "");
+						}
+					}
+				}
+			}
 		}
 	}
-
-	private void updateHighlightList()
+	
+	@Subscribe
+	public void onClanMemberJoined(ClanMemberJoined event)
+	{
+		getCallerList();
+	}
+	
+	@Subscribe
+	public void onClanMemberLeft(ClanMemberLeft event)
+	{
+		getCallerList();
+	}
+	
+	public void getCallerList()
 	{
-		highlightedPlayers.clear();
-		for (String player : COMMA_SPLITTER.splitToList(config.getHighlightedNames().toLowerCase().trim()))
+		callers.clear();
+		if (config.useClanchatRanks() && client.getClanMembers() != null)
 		{
-			highlightedPlayers.put(player, null);
+			for (ClanMember clanMember : client.getClanMembers())
+			{
+				if (clanMember.getRank().getValue() > config.callerRank().getValue())
+				{
+					callers.add(clanMember.getUsername());
+				}
+			}
 		}
+		if (config.callers().contains(","))
+		{
+			callers.addAll(Arrays.asList(config.callers().split(",")));
+		}
+		else
+		{
+			if (!config.callers().equals("") || config.callers().length() > 1)
+			{
+				callers.add(config.callers());
+			}
+		}
+		pileList = Arrays.asList(new String[callers.size()]);
 	}
-
-	boolean isHighlighted(Actor player)
+	
+	public boolean isCaller(Player player)
 	{
-		for (Map.Entry<String, Actor> map : highlightedPlayers.entrySet())
+		if (callers != null)
 		{
-			if (WildcardMatcher.matches(map.getKey(), player.getName()))
+			for (String name:callers)
 			{
-				return true;
+				String finalName = name.toLowerCase().replace("_", " ");
```

Comment:
this'll be the last Text.toJagexName() I comment, I swear (do it everywhere tho ;))